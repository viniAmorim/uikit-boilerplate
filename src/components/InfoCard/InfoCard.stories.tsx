import React from 'react';

import { Story, Meta } from '@storybook/react';

import InfoCard, { InfoCardProps } from './InfoCard';

export default {
  title: 'Atoms/Cards/InfoCard',
  component: InfoCard,
} as Meta;

const Template: Story<InfoCardProps> = (args) => (
  <div style={{ width: '300px' }}>
    <InfoCard {...args} />
  </div>
);

export const Default = Template.bind({});
Default.args = {
  title: 'Robert Santos',
  info: 12,
};
