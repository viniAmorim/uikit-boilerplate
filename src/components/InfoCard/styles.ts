import styled, { css } from 'styled-components';

export const Card = styled.div`
  width: 100%;
  height: 100%;
  min-width: 265px;
  display: flex;
  flex-direction: column;
  border-radius: 8px;
  box-sizing: border-box;
  padding: 19px;
  gap: 8px;
  border: 1px solid ${({ theme }) => theme.colors.grey100};
  box-shadow: 0px 7px 7px -5px rgba(0, 0, 0, 0.04),
    0px 10px 15px -5px rgba(0, 0, 0, 0.05), 0px 1px 3px rgba(0, 0, 0, 0.05);
  word-wrap: break-word;

  ${({ onClick }) =>
    onClick &&
    css`
      cursor: pointer;
    `}
`;
