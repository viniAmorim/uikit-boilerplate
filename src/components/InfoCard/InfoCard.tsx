import React, { forwardRef } from 'react';

import Text from '../Text';
import * as Styled from './styles';

export interface InfoCardProps extends React.HTMLAttributes<HTMLDivElement> {
  title: string;
  info: number;
  onClick?: () => void;
}

const UserCard = forwardRef<HTMLDivElement, InfoCardProps>(function Card(
  { title, info, onClick, ...rest },
  ref,
) {
  return (
    <Styled.Card onClick={onClick} ref={ref} {...rest}>
      <Text fontSize="14px" fontColor="grey500" weight="400" lineHeight="22px">
        {title}
      </Text>
      <Text
        fontSize="40px"
        fontColor="purple900"
        weight="700"
        lineHeight="62px"
      >
        {info.toLocaleString('pt-BR')}
      </Text>
    </Styled.Card>
  );
});

export default UserCard;
