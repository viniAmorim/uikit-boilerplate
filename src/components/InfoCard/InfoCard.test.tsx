import React from 'react';

import { render } from '~/test/utils';

import InfoCard from './InfoCard';

describe('InfoCard', () => {
  it('should render correctly', () => {
    const { container, getByText } = render(
      <InfoCard title="Park Shopping" info={121212} />,
    );

    const cardElement = container.firstChild as HTMLElement;

    expect(getByText('Park Shopping')).toBeInTheDocument();
    expect(getByText('121.212')).toBeInTheDocument();
    expect(cardElement).toBeInTheDocument();
    expect(cardElement).toMatchSnapshot();
  });
});
