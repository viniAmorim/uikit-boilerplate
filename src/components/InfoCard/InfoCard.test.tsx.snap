// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard should render correctly 1`] = `
<div
  class="sc-gueYoa fThvOw"
>
  <div
    class="sc-beqWaB bWlFyq"
    font-size="14px"
  >
    Park Shopping
  </div>
  <div
    class="sc-beqWaB iIaycp"
    font-size="40px"
  >
    121.212
  </div>
</div>
`;
